apps/protocol-frontend-e2e/src/support/commands.ts

```diff
@@ -41,4 +41,140 @@ Cypress.Commands.add('interceptGQL', (httpMethod, operationNames) => {
+    cy.fixture('contributions.json').then((contributions) => {
+      const contribution = contributions[0]
+      const mintedContributionQuery = `
+      INSERT INTO "Contribution" (id, name, status_id, activity_type_id, user_id, date_of_engagement, details, proof )
+        VALUES (1000, '${contribution.name}',2 , 1, 100, current_timestamp,
+         '${contribution.details}', '${contribution.proof}'
+        )
+      ON CONFLICT DO NOTHING;
+      `
+      cy.task('queryDatabase', mintedContributionQuery); 
+    });
+  }
+  else if (tableName=="GuildContribution"){
+
+    const GuildContributionQuery = `
+    INSERT INTO "GuildContribution" (guild_id, contribution_id )
+      VALUES (1, 1)
+    ON CONFLICT DO NOTHING;
+    `
+    cy.task('queryDatabase', GuildContributionQuery); 
+    
+  }
+  else if (tableName=="ContributionStatus"){
+
+    const GuildContributionQuery = `
+    INSERT INTO "ContributionStatus" (name) VALUES 
+      ('staging'),
+      ('minted');
+    `
+    cy.task('queryDatabase', GuildContributionQuery); 
+    
+  }
+  else if (tableName=="LoginUser1" || tableName=="LoginUser2"){
+    cy.fixture('users.json').then((accounts) => {
+      let userAccount: { username: string; address: string; email: string; };
+      if (tableName[tableName.length-1] == "1"){ //user1
+        userAccount = accounts[1]
+      }else{
+        userAccount = accounts[0]
+      }
+      const User1Login = `
+      INSERT INTO "User" (name, address, chain_type_id, active, email )
+        VALUES ('${userAccount.username}', '${userAccount.address}', 
+          1, true, '${userAccount.email}'
+        )
+      ON CONFLICT DO NOTHING;
+      `
+      cy.task('queryDatabase', User1Login);
+    });
+  }
+  else if (tableName=="User1CreateAndMintContribution"){
+    cy.fixture('contributions.json').then((contributions) => {
+      const user1ContributionData = contributions[3]
+      const User1CreateAndMintContribution = `
+      INSERT INTO "Contribution" (name, status_id, activity_type_id, user_id, date_of_engagement, details, proof )
+        VALUES ('${user1ContributionData.name}',2 , 1, 1, current_timestamp,
+         '${user1ContributionData.details}', '${user1ContributionData.proof}'
+        )
+      ON CONFLICT DO NOTHING;
+      `
+      cy.task('queryDatabase', User1CreateAndMintContribution);
+    });
+  }
+  else if (tableName=="GuildUser1" || tableName=="GuildUser2"){
+    let user_id: number
+    if (tableName[tableName.length-1] == "1"){
+      user_id=1
+    }else{
+      user_id=2;
+    }
+      const GuildUser1Query = `
+      INSERT INTO "GuildUser" (user_id, guild_id)
+        VALUES (${user_id}, 1)
+      ON CONFLICT DO NOTHING;
+      `
+      cy.task('queryDatabase', GuildUser1Query );
+  }
+
+});
+
+// tearDown DB
+Cypress.Commands.add('teardownDB',(tableNames)=>{
+    for (const tableName of tableNames){
+      const teardownQuery = `
+        TRUNCATE TABLE "${tableName}" RESTART IDENTITY CASCADE;
```

Comment:
We don't want to truncate the database. This will delete data that was created outside of the tests. Instead we want to be very specific with what we delete. And it is okay to have a different teardown for each test. We don't need a general teardown function

Reply:
Yeah, this makes sense! Will fix this in a moment